### PR TITLE
check if a page has uneven width or height and raise an issue

### LINKED ISSUE #250: mp4 height not divisible by 2....
it seems like the width and height must be divisible by 2

```traceback
[libx264 @ 0x7fbc88000c00] height not divisible by 2 (842x595)
Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
```

producing 
```traceback
RuntimeError: 'ffmpeg' failed with error code 1
```

https://stackoverflow.com/questions/20847674/ffmpeg-libx264-height-not-divisible-by-2/24260663

thank  @sarahtilley

### PR DESCRIPTION
**Pull request by typemytype**
+ test

fixes #250

### CODE REVIEW

File: tests/testExport.py [hunk 1]
```diff
@@ -262,6 +262,13 @@ def test_saveImage_multipage_keywordArgument(self):
                 drawBot.saveImage(tmp.path, multipage=False)
         self.assertEqual(output.lines(), [])
 
+    def test_unevenPages_mp4(self):
+        self.makeTestAnimation(pageWidth=500, pageHeight=501)
```
Comment (justvanrossum) [inline tests/testExport.py]: We should also test odd width and even height. Perhaps just generate a single frame?

File: drawBot/context/mp4Context.py [hunk 2]
```diff
@@ -37,6 +39,14 @@ def _writeDataToFile(self, data, path, options):
         if len(frameDurations) > 1:
             warnings.warn("Exporting to mp4 doesn't support varying frame durations, only the first value was used.")
 
+        pdfDocument = Quartz.PDFDocument.alloc().initWithData_(data)
```
[the remark this answers is not in the document]
Comment (justvanrossum) [reply drawBot/context/mp4Context.py]: Sorry, I didn't understand this correctly. (Thinking.)

File: drawBot/context/imageContext.py [hunk 3]
```diff
@@ -99,6 +102,9 @@ def _writeDataToFile(self, data, path, options):
                 page = pdfDocument.pageAtIndex_(index)
                 image = AppKit.NSImage.alloc().initWithData_(page.dataRepresentation())
                 imageRep = _makeBitmapImageRep(image, imageResolution)
+                if self.ensureEvenPixelDimensions:
+                    if imageRep.pixelsWide() % 2 or imageRep.pixelsHigh() % 2:
+                        raise DrawBotError("Exporting to %s doesn't support uneven page width and height." % (", ".join(self.fileExtensions)))
```
Comment (justvanrossum) [inline drawBot/context/imageContext.py]: uneven -> odd

Comment (justvanrossum) [reply drawBot/context/imageContext.py]:
```python
"Exporting to %s doesn't support odd pixel dimensions for width and height."
```

File: tests/testExport.py [hunk 4]
```diff
@@ -262,6 +262,13 @@ def test_saveImage_multipage_keywordArgument(self):
                 drawBot.saveImage(tmp.path, multipage=False)
         self.assertEqual(output.lines(), [])
 
+    def test_unevenPages_mp4(self):
```
[the remark this answers is not in the document]
Comment (justvanrossum) [reply tests/testExport.py]: No, simply as a comment in the test code.

[the remark this answers is not in the document]
Comment (typemytype) [reply tests/testExport.py]: Added

File: tests/testExport.py (hunk 1 above)
Comment (typemytype) [reply tests/testExport.py]: This is actually tested in other tests where an mp4 is generated with even dimensions.

Comment (justvanrossum) [reply tests/testExport.py]: I meant the other way: odd height, even width.

**Comment by typemytype** [comment]
This looks good to me!

**Pull request merged**